subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/espresso/action/TouchOutsideAction.kt

```diff
@@ -0,0 +1,88 @@
+package com.avito.android.lib.actions
+
+import android.graphics.Rect
+import android.os.SystemClock
+import android.view.InputDevice
+import android.view.MotionEvent
+import android.view.MotionEvent.PointerProperties
+import android.view.View
+import android.view.ViewConfiguration
+import android.view.ViewGroup
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.action.CoordinatesProvider
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.uiautomator.Configurator
+import androidx.test.uiautomator.UiDevice
+import org.hamcrest.Matcher
+
+class TouchOutsideAction : ViewAction {
+
+    override fun getConstraints(): Matcher<View> = ViewMatchers.isAssignableFrom(ViewGroup::class.java)
+
+    override fun getDescription(): String = "Press outside of dialog"
+
+    override fun perform(uiController: UiController, view: View) {
+        val coordinate = OutSideCoordinateProvider().calculateCoordinates(view)
+
+        val x = coordinate[0]
+        val y = coordinate[1]
+
+        val properties = PointerProperties()
+        properties.id = 0
+        properties.toolType = Configurator.getInstance().toolType
+
+        val coords = MotionEvent.PointerCoords()
+        coords.pressure = 1f
+        coords.size = 1f
+        coords.x = x
+        coords.y = y
+
+        val cur = SystemClock.uptimeMillis()
+        val events = listOf(
+            MotionEvent.obtain(
+                cur,
+                cur + 100,
+                MotionEvent.ACTION_DOWN, 1, arrayOf(properties), arrayOf(coords),
+                0, 0, 1.0f, 1.0f, 12, 0, InputDevice.SOURCE_TOUCHSCREEN, 0
+            ),
+            MotionEvent.obtain(
+                cur,
+                cur + 200,
+                MotionEvent.ACTION_UP, 1, arrayOf(properties), arrayOf(coords),
+                0, 0, 1.0f, 1.0f, 12, 0, InputDevice.SOURCE_TOUCHSCREEN, 0
+            )
+        )
+
+        view.rootView.let {
+            for (event in events) {
+                it.dispatchTouchEvent(event)
+            }
+        }
+
+        val duration: Int = ViewConfiguration.getPressedStateDuration()
+        // ensures that all work enqueued to process the tap has been run.
+        if (duration > 0) {
+            uiController.loopMainThreadForAtLeast(duration.toLong())
+        }
+    }
+
+}
+
+class OutSideCoordinateProvider : CoordinatesProvider {
+    override fun calculateCoordinates(view: View): FloatArray {
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val rect = Rect()
+        view.getGlobalVisibleRect(rect)
+
+        val y = uiDevice.displayHeight.toFloat() / 2
+        val x = uiDevice.displayWidth / 2f + rect.width() / 2f + HORIZONTAL_OFFSET_DP.dp
```

Comment:
Where will actually `x` be located on the screen?

Reply:
Added comment with explanation. Clicks to the left of the view. Actually HORIZONTAL_OFFSET_DP to the left